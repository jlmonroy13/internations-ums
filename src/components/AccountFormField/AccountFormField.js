import React from 'react';
import PropTypes from 'prop-types';

function AccountFormField({ iconName, id, ...restProps }) {
  return (
    <div className="account__form-group">
      <label className="account__label account__label--with-icon" htmlFor={id}>
        <span className={`icon icon--${iconName}`} />
      </label>
      <input className="account__form-control" id={id} {...restProps} />
    </div>
  );
}

AccountFormField.propTypes = {
  id: PropTypes.string.isRequired,
  iconName: PropTypes.string,
};

export default AccountFormField;
