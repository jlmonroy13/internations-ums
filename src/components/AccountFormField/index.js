import AccountFormField from './AccountFormField';

export default AccountFormField;
