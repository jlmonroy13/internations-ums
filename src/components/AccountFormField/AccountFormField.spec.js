import React from 'react';
import renderer from 'react-test-renderer';
import AccountFormField from './AccountFormField';

describe('<AccountFormField />', () => {
  it('should render the account form field component', () => {
    const wrapper = renderer.create(<AccountFormField id="email" iconName="user" />);
    expect(wrapper).toMatchSnapshot();
  });
});
