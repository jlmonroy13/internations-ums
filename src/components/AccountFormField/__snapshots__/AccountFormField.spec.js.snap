// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountFormField /> should render the account form field component 1`] = `
<div
  className="account__form-group"
>
  <label
    className="account__label account__label--with-icon"
    htmlFor="email"
  >
    <span
      className="icon icon--user"
    />
  </label>
  <input
    className="account__form-control"
    id="email"
  />
</div>
`;
